### ISSUE TITLE
xhyverun.sh: "hv_vm_create failed"

### ISSUE THREAD

**Issue by sometimesfood** [mist64/xhyve#18]
I am trying to test xhyve (commit 67d5ad8ee7203049f69d43138e219ac7aacea4bf) on a Mid 2010 MacBook Pro (MC374LL/A) running OS X 10.10.3 (14D136), but for some reason the `hv_vm_create` call in `vmx.c` seems to fail:

```
~/src/xhyve % ./xhyverun.sh 
hv_vm_create failed
./xhyverun.sh: line 15: 53220 Abort trap: 6           (core dumped) build/xhyve $MEM $SMP $PCI_DEV $LPC_DEV $NET $IMG_CD $IMG_HDD -f kexec,$KERNEL,$INITRD,"$CMDLINE"
```

My CPU (Core 2 Duo P8600 2.4 GHz) supports VT-x and it seems to be enabled properly (recent firmware, VMWare Fusion runs just fine).
```
~ % sysctl -a | grep machdep.cpu.features
machdep.cpu.features: FPU VME DE PSE TSC MSR PAE MCE CX8 APIC SEP MTRR PGE MCA CMOV PAT PSE36 CLFSH DS ACPI MMX FXSR SSE SSE2 SS HTT TM PBE SSE3 DTES64 MON DSCPL VMX SMX EST TM2 SSSE3 CX16 TPR PDCM SSE4.1 XSAVE
```

I am afraid the backtrace is not too useful. Is there any other information I can help you with?

**Comment by mist64** [comment]
Hypervisor.framework requires a CPU that supports EPT, i.e. hardware support for nested paging. The Core 2 does not have this feature, unfortunately. Maybe we could get a better error code from the framework and present it to the user...

**Comment by sometimesfood** [comment]
Right, I had already suspected my old CPU to be the root cause for this. Thanks for the clarification.

Apparently `hv_vm_create` returns `HV_NO_DEVICE` on my machine. I added an explicit check for `HV_NO_DEVICE` that prints an appropriate error message and returns the error from `vmx_init` and keeps the `xhyve_abort` for all other errors to a local feature branch.

Should I send you a pull request, or did you have a different course of action in mind?
